langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -1557,6 +1558,104 @@ function assert(anydata actual, anydata expected) {
     panic error(reason);
 }
 
+type RecordWithSimpleTypeFields record {|
+    int a;
+    byte b;
+    float c;
+    decimal d;
+    string e;
+    boolean f;
+|};
+
+type RecordWithArrayValueFields record {|
+    float[] a;
+    [string, int] t;
+|};
+
+type RecordAnydata record {|
+    int id;
+    anydata a;
+|};
+
+type RecordWithRecordField record {|
+    RecordAnydata r;
+|};
+
+type RecordWithMapField record {|
+    map<anydata> m;
+|};
+
+type RecordWithXmlField record {|
```

Comment:
Shall we have tests with open records & record with `anydata` rest fields?

Reply:
Did we add a test with rest fields? @gabilang

Reply:
It looks like there is an issue with the conversion of tables which contains records with rest fields as constrained type. Will address this issue separately and created an [issue](https://github.com/ballerina-platform/ballerina-lang/issues/32524) for that